# Phân tích dữ liệu Food Price In Turkey

Bộ dữ liệu FoodPrice_in_Turkey chứa dữ liệu chứa dữ liệu về giá cả các mặt hàng thực phẩm tại Turkey, các thông tin được thống kê bao gồm:
- Place: Nơi bán
- ProductID: Mã sản phẩm
- ProductName: Tên sản phẩm
- UmId: Mã đơn vị đo lường
- UmName: Tên đơn vị đo lường
- Month: Tháng
- Year: Năm
- Price: Giá 

Hãy đưa bộ dữ liệu vào phân tích và giải đáp các thắc mắc sau:

- Bộ dữ liệu chứa bao nhiêu dòng, bao nhiêu cột
- Thang đo tương ứng của các thuộc tính được lưu trữ
- Giá trung bình của từng loại thực phẩm là bao nhiêu
- Đề xuất những loại phân tích mà bạn muốn biết mà bộ dữ liệu này có thể cung cấp

## Import thư viên

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

## Đọc bộ dữ liệu FoodPrice

In [2]:
data = pd.read_csv('/content/drive/MyDrive/PhanTichDuLieu_CodeGym/Tuan01/data/FooPrice_in_Turkey.csv')
data.head(10)

Place  ProductId    ProductName  UmId UmName  Month  Year   Price
0  National Average         52  Rice - Retail     5     KG      5  2013  4.4920
1  National Average         52  Rice - Retail     5     KG      6  2013  4.5786
2  National Average         52  Rice - Retail     5     KG     11  2013  4.7865
3  National Average         52  Rice - Retail     5     KG     12  2013  5.1337
4  National Average         52  Rice - Retail     5     KG      1  2014  5.5099
5  National Average         52  Rice - Retail     5     KG      2  2014  5.8146
6  National Average         52  Rice - Retail     5     KG      3  2014  6.0014
7  National Average         52  Rice - Retail     5     KG      5  2014  6.3202
8  National Average         52  Rice - Retail     5     KG      6  2014  6.6019
9  National Average         52  Rice - Retail     5     KG      7  2014  6.6503

In [3]:
data.tail(10) # 10 dòng cuối cùng

Place  ProductId              ProductName  ...  Month  Year    Price
7371  Izmir        502  Cocoa (powder) - Retail  ...      3  2019  45.2125
7372  Izmir        502  Cocoa (powder) - Retail  ...      4  2019  45.4075
7373  Izmir        502  Cocoa (powder) - Retail  ...      5  2019  45.3125
7374  Izmir        502  Cocoa (powder) - Retail  ...      6  2019  46.4609
7375  Izmir        502  Cocoa (powder) - Retail  ...      7  2019  47.4010
7376  Izmir        502  Cocoa (powder) - Retail  ...      8  2019  47.9040
7377  Izmir        502  Cocoa (powder) - Retail  ...      9  2019  49.1176
7378  Izmir        502  Cocoa (powder) - Retail  ...     10  2019  50.8347
7379  Izmir        502  Cocoa (powder) - Retail  ...     11  2019  51.7985
7380  Izmir        502  Cocoa (powder) - Retail  ...     12  2019  51.6071

[10 rows x 8 columns]

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7381 entries, 0 to 7380
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Place        7381 non-null   object 
 1   ProductId    7381 non-null   int64  
 2   ProductName  7381 non-null   object 
 3   UmId         7381 non-null   int64  
 4   UmName       7381 non-null   object 
 5   Month        7381 non-null   int64  
 6   Year         7381 non-null   int64  
 7   Price        7381 non-null   float64
dtypes: float64(1), int64(4), object(3)
memory usage: 461.4+ KB


 Bộ dữ liệu gồm có 8 cột và 7381 dòng, Các thuộc tính định tính là Place, ProductName, Umname

In [5]:
data.describe()

ProductId         UmId        Month         Year        Price
count  7381.000000  7381.000000  7381.000000  7381.000000  7381.000000
mean    254.509687    15.252811     6.479881  2017.458474    15.738709
std     141.008599    25.276665     3.530361     1.499256    22.418284
min      52.000000     5.000000     1.000000  2013.000000     0.253900
25%     114.000000     5.000000     3.000000  2017.000000     3.223600
50%     239.000000     5.000000     6.000000  2018.000000     5.909600
75%     388.000000     5.000000    10.000000  2019.000000    20.441900
max     502.000000   102.000000    12.000000  2019.000000   159.445800

## Giá trị trung bình từng loại thực phẩm

In [6]:
price_mean = data.groupby(['ProductName'])['Price'].mean()
price_mean

ProductName
Apples (red) - Retail                                       2.392193
Apples - Retail                                             3.885039
Bananas - Retail                                            8.224784
Beans (white) - Retail                                      9.572830
Bread (common) - Retail                                     4.463293
Bread (pita) - Retail                                       2.404545
Bulgur - Retail                                             3.516712
Cabbage - Retail                                            1.723395
Cauliflower - Retail                                        3.274868
Cheese - Retail                                            25.522875
Chickpeas - Retail                                         10.347860
Cocoa (powder) - Retail                                    43.802339
Coffee (instant) - Retail                                  43.295493
Coffee - Retail                                            42.468075
Cucumbers (greenhouse)